slidesimport/slidesimport.py

```diff
@@ -127,5 +129,35 @@ def run(rawArgs=None):
         mediaFilePath = getMediaPath(collectionMediaPath, prefix, slideNum)
         mediaFileName = getMediaName(prefix, slideNum)
 
-        outputDeckFile.write('"{0}"; <img src="{1}" />\n'.format(cgi.escape(qs), mediaFileName))
+        fallback_flag = False
+
+        # If the "question without slide" dict entry is not empty, write that
+        if notesQWS[slideNum] != '':
+            outputDeckFile.write('"{0}"; '.format(cgi.escape(notesQWS[slideNum])))
+        # Otherwise try "question followed by slide"
+        elif notesQFBS[slideNum] != '':
+            outputDeckFile.write('<div>{0}</div><img src="{1}" />; '.format(cgi.escape(notesQFBS[slideNum]), mediaFileName))
+        # Otherwise try "slide followed by question"
+        elif notesSFBQ[slideNum] != '':
+            outputDeckFile.write('<img src="{0}" /><div>{1}</div>; '.format(mediaFileName, cgi.escape(notesSFBQ[slideNum])))
+        else:
+            fallback_flag = True
+
+        # If the "answer without slide" dict entry is not empty, write that
+        if notesAWS[slideNum] != '':
+            outputDeckFile.write('"{0}"\n'.format(cgi.escape(notesAWS[slideNum])))
+        # Otherwise try "answer followed by slide"
+        elif notesAFBS[slideNum] != '':
+            outputDeckFile.write('<div>{0}</div><img src="{1}" />\n'.format(cgi.escape(notesAFBS[slideNum]), mediaFileName))
+        # Otherwise try "slide followed by answer"
+        elif notesSFBA[slideNum] != '':
+            outputDeckFile.write('<img src="{0}" /><div>{1}</div>\n'.format(mediaFileName, cgi.escape(notesSFBA[slideNum])))
+        else:
+            fallback_flag = True
```

Comment:
The `;` delimiters are actually present because they always get written from the question side before the answer part is written.  I actually rewrote this part of the code anyway, because the way I was doing the `fallback_flag` thing was causing a mild bug whereby in certain cases, i.e. when just a `Q: ` was present, it would set the fallback flag from the answer section, and end up writing out the `Q: ` along with the full question as part of the fallback behaviour.  In any case, everything should be working properly now!